test/integration-test/src/utils.rs

```diff
@@ -0,0 +1,76 @@
+//! Utilities to run tests
+
+use std::{
+    fs::File,
+    io::Write,
+    os::fd::AsRawFd,
+    process::{Command, Stdio},
+    sync::atomic::{AtomicU64, Ordering},
+};
+
+use libc::{setns, CLONE_NEWNET};
+
+fn script(script: &str) {
+    // We pipe stdout/err to capture it and print it ourselves, so that it gets captured by the
+    // test harness
+    let mut bash = Command::new("/bin/bash")
+        .arg("-e")
+        .stdin(Stdio::piped())
+        .spawn()
+        .unwrap();
+
+    // Send script
+    let mut stdin = bash.stdin.take().unwrap();
+    stdin.write_all(script.as_bytes()).unwrap();
+    stdin.write_all(b"\n").unwrap();
+    drop(stdin);
```

Comment:
Why the explicit drop of stdin?

Reply:
Otherwise stdin does not get closed, and the bash we execute does not close terminate due to a missing EOF, deadlocking in the process.

Reply:
Resolving conversation as this code has been dropped.